src/layer-composer/generators/vessel-events/vessel-events.ts

```diff
@@ -62,27 +109,32 @@ class VesselsEventsGenerator {
     return layers
   }
 
-  getStyle = (layer: VesselEventsGeneratorConfig) => {
+  getStyle = (config: VesselEventsGeneratorConfig) => {
     return {
-      id: layer.id,
-      sources: this._getStyleSources(layer),
-      layers: this._getStyleLayers(layer),
+      id: config.id,
+      sources: this._getStyleSources(config),
+      layers: this._getStyleLayers(config),
     }
   }
 }
 
 export default VesselsEventsGenerator
 
 type AuthorizationOptions = 'authorized' | 'partially' | 'unmatched'
-type Event = {
+
+type RawEvent = {
+  id: string
   type: string
-  active: boolean
-  timestamp: number
-  authorizationStatus: AuthorizationOptions
-  coordinates: {
-    lng: number
+  position: {
+    lng?: number
+    lon?: number
     lat: number
   }
+  start: number
+  encounter?: {
+    authorized: boolean
+    authorizationStatus: AuthorizationOptions
```

Comment:
this is directly related with the API response, do you think worth it to keep it synced using the same types ? however how to reuse types across projects is something we have to discuss I'd like to open the conversation.